src/omics/ugbio_omics/omics_cache_path.py

```diff
@@ -0,0 +1,120 @@
+
+    # Retrieve the single subfolder
+    subfolder = response["CommonPrefixes"][0]["Prefix"]
+    # Construct and return the full S3 URI
+    return f"s3://{bucket}/{subfolder}"
+
+
+def get_run_cache_path(run_id, task_id=None):
+    omics_client = get_aws_client("omics")
+
+    omics_run = omics_client.get_run(id=run_id)
+    logging.debug(omics_run)
+    cache_id = omics_run["cacheId"]
+
+    omics_cache = omics_client.get_run_cache(id=cache_id)
+    logging.debug(omics_cache)
+    cache_base_uri = f"{omics_cache['cacheS3Uri']}{run_id}"
+    if not task_id:
+        logging.info(f"Run cache uri: {cache_base_uri}")
+        cache_uri = cache_base_uri
+    else:
+        task_cache_uri = f"{cache_base_uri}/{task_id}/"
+        full_cache_uri = build_cache_full_uri(task_cache_uri)
+        logging.info(f"Run task cache uri: {full_cache_uri}")
+        cache_uri = full_cache_uri
+    return cache_uri
+
+
+def get_aws_client(service):
+    return boto3.client(service)
+
+
+def list_objects_with_index(s3_client, bucket_name, prefix):
+    """
+    List all objects under the given S3 prefix that contain '_index' in their keys.
+    """
+    paginator = s3_client.get_paginator("list_objects_v2")
+    for page in paginator.paginate(Bucket=bucket_name, Prefix=prefix):
+        for obj in page.get("Contents", []):
+            key = obj["Key"]
+            if "_index" in key:
+                yield key
+
+
+def copy_object(s3_client, bucket_name, source_key, dest_key):
+    """
+    Copy an S3 object from source_key to dest_key within the same bucket.
+    """
+    copy_source = {"Bucket": bucket_name, "Key": source_key}
+    s3_client.copy(copy_source, bucket_name, dest_key)
+    logging.info(f"Copied {source_key} to s3://{bucket_name}/{dest_key}")
+
+
+def process_s3_uri(s3_uri):
+    """
+    Parse the S3 URI and return the bucket name and prefix.
+    """
+    parsed = urlparse(s3_uri)
+    if parsed.scheme != "s3":
+        raise ValueError(f"Invalid S3 URI: {s3_uri}")
+    return parsed.netloc, parsed.path.lstrip("/")
+
+
+def copy_omics_cached_indexes(cache_s3_uri):
+    # Initialize S3 client
+    s3_client = boto3.client("s3")
+
+    # Parse bucket and prefix
+    bucket_name, prefix = process_s3_uri(cache_s3_uri)
+
+    # List and process objects
+    for source_key in list_objects_with_index(s3_client, bucket_name, prefix):
+        # Create destination key by removing '_index' substring
+        dest_key = re.sub(r"_index", "", source_key)
+        copy_object(s3_client, bucket_name, source_key, dest_key)
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("run_id", help="AWS HealthOmics run id")
+    parser.add_argument(
+        "--task-id", type=str, help="HealthOmics run task-id. Leave empty to get the base run cache uri."
+    )
+    parser.add_argument("--copy-indexes", dest="copy_indexes", action="store_true", help="copy indexes cached files")
+    parser.set_defaults(copy_indexes=False)
```

Comment:
can it copy indexes by default for all tasks? I usually don't know which files I need to check when I run something, so setting it to false by default will not be so helpful
Also in the github action by UI which is the way I use to run things
Thanks

Reply:
it'll copy the indexes for all the tasks by default
Only if you add --task-id it'll copy the indexes of that specific task only

what you need is to specify the run-id and add --copy_indexes
`omics_cache_path <RUN_ID> --copy-indexes`

Reply:
ok cool, and in the github action UI it will copy them all by default?

Reply:
no, as I said we decided to not call this automatically 
When you have run that you want its cache indexes to be copied call this script

Reply:
oh, I understand. To be honest I think I will just copy the index file manually then.. finding the docker and finding the name of the command and creating the command will take me more time that just do that manually. I just wanted to make the way of working with omics easier by getting files that I just can load into igv or use with samtools,  but it does not so help.. Sorry
You can merge that on my side, but I am pretty sure I will not use that

Reply:
the omics docker is always "latest" and the command also doesn't changed
you can write an alias in your bachrc to run it easy:
```
copy_cache_indexes () {
	run=${1}
	docker run -v ~/.aws:/root/.aws -e AWS_PROFILE=$AWS_PROFILE 337532070941.dkr.ecr.us-east-1.amazonaws.com/ugbio_omics:latest omics_cache_path --run-id $run --copy-indexes
}
```

Reply:
ok looks better, thanks